quint/src/runtime/impl/compilerImpl.ts

```diff
@@ -35,9 +35,12 @@ import * as ir from '../../ir/quintIr'
 import { RuntimeValue, rv } from './runtimeValue'
 import { ErrorCode } from '../../quintError'
 
-import { lastTraceName } from '../compile'
+import { inputDefName, lastTraceName } from '../compile'
 import { unreachable } from '../../util'
 
+// Internal names in the compiler, which have special treatment.
+// For some reason, if we replace 'q::input' with inputDefName, everything breaks.
```

Comment:
I'm curious, what breaks specifically?

Reply:
Plenty of tests. I tried to understand, how this was possible, but could not. It really looked like magic to me